### PR TITLE
ADD json info to the logs

### LINKED ISSUE #3: Log relevant information in JSON format
Current logs have this format:

```
Response with status 304 in 5 ms
```

They are not as useful as they should for a machine. What do you thing about logging them as:

```
Response {"statusCode": 304, "duration": 5}
```

I think we just need to change:

``` js
 logger.info('Response with status %d in %d ms.', res.statusCode, duration);
```

to:

``` js
 logger.info('Response', {statusCode: res.statusCode, duration: duration});
```

and would work for both [logops](https://github.com/telefonicaid/logops) and no-logops users

### PR DESCRIPTION
**Pull request by palmerabollo**
Fixes #3, adding JSON fields to be able to work with them (e.g. calculate the average response time or the number of requests with response code between 400-499).

/cc @jlorgal @spmarin what do you think? The only issue I can think about is that it could break tests that check the exact format of the logs.

### CODE REVIEW

File: lib/logging.js [hunk 1]
```diff
@@ -54,15 +54,30 @@ module.exports = function(logger) {
...
-    logger.info('Request from %s: %s %s', getClientIp(req), req.method, req.originalUrl);
+    var clientIp = getClientIp(req);
+
+    var reqInfo = {
+      requestClientIp: clientIp,
+      requestMethod: req.method,
+      requestUrl: req.originalUrl
+    };
+
+    logger.info('Request from %s: %s %s', getClientIp(req), req.method, req.originalUrl, reqInfo);
```
Comment (jlorgal) [inline lib/logging.js:65]: What about simplify with:
logger.info('Request %s %s', req.method, req.originalUrl, reqInfo);
?

File: lib/logging.js [hunk 2]
```diff
@@ -54,15 +54,30 @@ module.exports = function(logger) {
...
 
     onHeaders(res, function onResponse() {
       var duration = new Date() - startTime;
       var location = res.get('location');
+
+      var resInfo = {
```
Comment (jlorgal) [inline lib/logging.js:71]: What about?

```
var resInfo = {
  responseStatusCode: res.statusCode,
  responseDuration: duration,
  responseLocation: res.get('location')
};
logger.info('Response with status %d', res.statusCode, resInfo);
```

File: lib/logging.js [hunk 3]
```diff
@@ -54,15 +54,30 @@ module.exports = function(logger) {
 
   return function loggingMiddleware(req, res, next) {
     var startTime = new Date();
-    logger.info('Request from %s: %s %s', getClientIp(req), req.method, req.originalUrl);
+    var clientIp = getClientIp(req);
+
+    var reqInfo = {
+      requestClientIp: clientIp,
```
Comment (jlorgal) [inline lib/logging.js:60]: What's the impact of long keys in terms of performance of Iris?

File: lib/logging.js (hunk 1 above)
Comment (palmerabollo) [inline lib/logging.js:65]: ok

File: lib/logging.js (hunk 2 above)
Comment (palmerabollo) [inline lib/logging.js:71]: ok

File: lib/logging.js (hunk 3 above)
Comment (palmerabollo) [inline lib/logging.js:60]: no idea (yet)

**Comment by jlorgal** [comment]
https://github.com/Telefonica/node-express-logging/pull/6 is an alternative to this PR. Close it?

**Pull request closed without merging**